File: csharp/allergies/Allergies.cs
using System;
using System.Linq;

public enum Allergen
{
    Eggs = 1,
    Peanuts = 2,
    Shellfish = 4,
    Strawberries = 8,
    Tomatoes = 16,
    Chocolate = 32,
    Pollen = 64,
    Cats = 128
}

public class Allergies
{
    private readonly int _mask;

    public Allergies(int mask) => _mask = mask;

    public bool IsAllergicTo(Allergen allergen) => (_mask & (int)allergen) > 0;

    public Allergen[] List() => Enum.GetValues(typeof(Allergen)).Cast<Allergen>().Where(IsAllergicTo).ToArray();
}

File: swift/allergies/Sources/Allergies/Allergies.swift
class Allergies {
  let allergens = [
    "eggs": 1,
    "peanuts": 2,
    "shellfish": 4,
    "strawberries": 8,
    "tomatoes": 16,
    "chocolate": 32,
    "pollen": 64,
    "cats": 128,
  ]

  private let mask: Int

  init(_ mask: Int) {
    self.mask = mask
  }

  func allergicTo(item: String) -> Bool {
    if let value = allergens[item] {
      return self.mask & value == value
    }

    return false
  }

  func list() -> [String] {
    allergens
      .sorted(by: { $0.value < $1.value })
      .filter { allergicTo(item: $0.key) }
      .map { $0.key }
  }
}
